archethic-foundation/archethic-node
# Fix authorized nodes unavailability

# Description

This PR aims to resolve issue regarding nodes which become unavailable and try to rejoin later while the node shared secrets have been performed since.

But once the node is coming back it should be able to load missed transactions without issue regarding election or shared secrets

Fixes #318 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- Start few nodes and wait they become authorized
- Stop one
- Wait this node becomes unavailable
- Wait other nodes perform shared secrets transactions
- Start the paused node
- It should be able to decode the missed transactions and start validate transaction again


# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules
